### PR TITLE
Add some cli args, unify additional_cache

### PR DESCRIPTION
**Pull request by charleskawczynski**
I think this proposed pattern is a bit of a balance:

```
additional_cache(Y, params, dt; use_tempest_mode) = merge(
    hyperdiffusion_cache(Y; κ₄ = FT(2e17), use_tempest_mode),
    sponge ? rayleigh_sponge_cache(Y, dt) : NamedTuple(),
    isnothing(micro) ? NamedTuple() : zero_moment_microphysics_cache(Y),
    isnothing(HSF) ? NamedTuple() : held_suarez_cache(Y),
    isnothing(rad) ? NamedTuple() : rrtmgp_model_cache(Y, params; idealized_h2o),
    isnothing(turbconv) ? NamedTuple() : vertical_diffusion_boundary_layer_cache(Y),
)
```
We still use a closure, but everything that's included is kind of directly visible at the top level. Thoughts, @dennisYatunin, @jiahe23 ?

I haven't updated the cli options, yet, since they were being updated in #397.

### CODE REVIEW

File: examples/hybrid/cli_options.jl [hunk 1]
```diff
@@ -24,6 +24,20 @@ function parse_commandline()
         help = "Moisture model [`dry` (default), `equil`, `non_equil`]"
         arg_type = String
         default = "dry"
+        "--micro"
```
Comment (jiahe23) [inline examples/hybrid/cli_options.jl:27]: I would call it `microphy`

File: examples/hybrid/cli_options.jl [hunk 2]
```diff
@@ -24,6 +24,20 @@ function parse_commandline()
...
+        help = "Microphysics model [`nothing` (default), `0M`]"
+        arg_type = String
+        "--HSF"
```
Comment (jiahe23) [inline examples/hybrid/cli_options.jl:30]: Can we have a `---forcing` default to none and make held-suarez forcing an option here?

File: examples/hybrid/cli_options.jl [hunk 3]
```diff
@@ -24,6 +24,20 @@ function parse_commandline()
...
+        help = "Held Suarez Forcing [`false` (default), `true`]"
+        arg_type = Bool
+        default = false
+        "--turbconv"
+        help = "Turbulence Convection model [`nothing` (default), `const_diffusivity`]"
+        arg_type = String
+        "--idealized_h2o"
```
Comment (jiahe23) [inline examples/hybrid/cli_options.jl:37]: Can we make this an option in `--rad`

**Comment by jiahe23** [comment]
I left a few comments. Also, don't forget to modify the README

File: examples/hybrid/cli_options.jl (hunk 2 above)
Comment (charleskawczynski) [reply examples/hybrid/cli_options.jl:30]: Yesss, thank you!

File: examples/hybrid/cli_options.jl (hunk 3 above)
Comment (charleskawczynski) [reply examples/hybrid/cli_options.jl:37]: Sounds good to me, how can we do that?

Comment (charleskawczynski) [reply examples/hybrid/cli_options.jl:37]: I'm not sure how to do this off the top of my head, so I'm going to save this for a separate PR.

**Comment by charleskawczynski** [comment]
I forgot the readme. Fixed now.

File: examples/hybrid/cli_options.jl (hunk 3 above)
Comment (jiahe23) [reply examples/hybrid/cli_options.jl:36]: might need some inputs from zhaoyi and dennis

**Pull request merged**